lib/rules/string.js

```diff
@@ -0,0 +1,392 @@
+import SyncRule from '../types/syncRule';
+import FlagRule from '../types/flagRule';
+import url from 'url';
+import { escapeRegExp } from '../util';
+import isURL from 'validator/lib/isURL';
+
+/* eslint-disable max-len */
+const emailRegex = /^[-a-z\d~!$%^&*_=+}{\'?]+(\.[-a-z\d~!$%^&*_=+}{\'?]+)*@([a-z\d_][-a-z\d_]*(\.[-a-z\d_]+)*\.([a-zрф]{2,6})|([\d]{1,3}\.[\d]{1,3}\.[\d]{1,3}\.[\d]{1,3}))(:[\d]{1,5})?$/i;
+const guidRegex = /^[\dA-F]{8}[-]?([\dA-F]{4}[-]?){3}[\dA-F]{12}$/i;
+const hexRegex = /^[a-f\d]+$/i;
+
+const dateRegex = /^([\+-]?\d{4}(?!\d{2}\b))((-?)((0[1-9]|1[0-2])(\3([12]\d|0[1-9]|3[01]))?|W([0-4]\d|5[0-2])(-?[1-7])?|(00[1-9]|0[1-9]\d|[12]\d{2}|3([0-5]\d|6[1-6])))([T\s]((([01]\d|2[0-3])((:?)[0-5]\d)?|24\:?00)([\.,]\d+(?!:))?)?(\17[0-5]\d([\.,]\d+)?)?([zZ]|([\+-])([01]\d|2[0-3]):?([0-5]\d)?)?)?)?$/;
+const ccRegex = /^[\d]{16}$/i;
+const alnumRegex = /^[a-z\d]+$/i;
+const tokenRegex = /^\w+$/i;
+const ipv4Regex = /^(:?25[0-5]|2[0-4]\d|[01]?\d+)\.(:?25[0-5]|2[0-4]\d|[01]?\d+)\.(:?25[0-5]|2[0-4]\d|[01]?\d+)\.(:?25[0-5]|2[0-4]\d|[01]?\d+)(\/\d+)?$/i;
+const ipv6RegEx = /(:?[\da-f]{1,4}:){7,7}[\da-f]{1,4}|(:?[\da-f]{1,4}:){1,7}:|(:?[\da-f]{1,4}:){1,6}:[\da-f]{1,4}|(:?[\da-f]{1,4}:){1,5}(:?:[\da-f]{1,4}){1,2}|(:?[\da-f]{1,4}:){1,4}(:?:[\da-f]{1,4}){1,3}|(:?[\da-f]{1,4}:){1,3}(:?:[\da-f]{1,4}){1,4}|(:?[\da-f]{1,4}:){1,2}(:?:[\da-f]{1,4}){1,5}|[\da-f]{1,4}:(:?(:?:[\da-f]{1,4}){1,6})|:(:?(:?:[\da-f]{1,4}){1,7}|:)|fe80:(:?:[\da-f]{0,4}){0,4}%[\da-zA-Z]{1,}|::(:?ffff(:?:0{1,4}){0,1}:){0,1}(:?(:?25[0-5]|(:?2[0-4]|1{0,1}[\d]){0,1}[\d])\.){3,3}(:?25[0-5]|(:?2[0-4]|1{0,1}[\d]){0,1}[\d])|(:?[\da-f]{1,4}:){1,4}:(:?(:?25[0-5]|(:?2[0-4]|1{0,1}[\d]){0,1}[\d])\.){3,3}(:?25[0-5]|(:?2[0-4]|1{0,1}[\d]){0,1}[\d])(\/d+)?/i;
+const hostnameRegEx = /^(([a-z\d]|[a-z\d][a-z\d\-]*[a-z\d])\.)*([a-z\d]|[a-z\d][a-z\d\-]*[a-z\d])$/i;
```

Comment:
I think it'd be fine to pull these out of validator.js. You get some goodies like the Luhn formula for credit card numbers, and is code we don't have to explicitly test/maintain. True it's more filesize, but it's good functionality and validator.js is pretty compact for what it does.